src/mrb_json.c

```diff
@@ -24,12 +24,7 @@
  *********************************************************/
 static mrb_bool
 mrb_method_defined(mrb_state* mrb, mrb_value value, const char* name) {
-  int ai = mrb_gc_arena_save(mrb);
-  mrb_sym mid = mrb_intern_cstr(mrb, name);
-  mrb_value methods = mrb_funcall(mrb, value, "public_methods", 1, mrb_false_value());
-  mrb_value included = mrb_funcall(mrb, methods, "include?", 1, mrb_symbol_value(mid), NULL);
-  mrb_gc_arena_restore(mrb, ai);
-  return mrb_type(included) == MRB_TT_TRUE;
+  return mrb_respond_to(mrb, value, mrb_intern_cstr(mrb, name));
```

Comment:
@mattn it seems like first version used `mrb_obj_respond_to` and had issues, which now seems to be resolved on mruby master, and the code works as expected, while return in `include?` causes `LocalJumpError`.